packages/snowpack/src/logger.ts

```diff
@@ -1,35 +1,88 @@
 import * as colors from 'kleur/colors';
-import pino, {LoggerOptions} from 'pino';
-
-const LEVEL = {
-  TRACE: 10,
-  DEBUG: 20,
-  INFO: 30,
-  WARN: 40,
-  ERROR: 50,
-  FATAL: 60,
-};
+import {LoggerLevel, LoggerEvent, LoggerOptions} from './types/snowpack';
 
-/** http://getpino.io/#/docs/pretty */
-export function prettifier(options: LoggerOptions) {
-  return (inputData: pino.LogDescriptor) => {
-    // if the log starts with brackets, use that (e.g. [@snowpack/plugin-babel]); otherwise, use [snowpack]
-    const name = options.name ? `[${options.name}]` : '[snowpack]';
-    let msg = `${colors.dim(name)} ${inputData.msg}\n`; // add newline at end
+const levels: Record<LoggerLevel, number> = {
+  debug: 20,
+  info: 30,
+  warn: 40,
+  error: 50,
+  silent: 90,
+};
 
-    if (inputData.level === LEVEL.ERROR || inputData.level === LEVEL.FATAL) msg = colors.red(msg);
-    if (inputData.level === LEVEL.WARN) msg = colors.yellow(msg);
+/** Custom logger heavily-inspired by https://github.com/pinojs/pino with extra features like log retentian */
+class SnowpackLogger {
+  /** set the log level (can be changed after init) */
+  public level: LoggerLevel = 'info';
 
-    return msg;
+  private history: string[] = []; // this is immutable; must be accessed with Logger.getHistory()
```

Comment:
see comment above about concerns about this getting too large and slowing dev to a crawl.